packages/paste-core/components/checkbox/stories/checkbox.stories.tsx

```diff
@@ -13,8 +13,18 @@ export default {
 };
 
 export const DefaultCheckbox = (): React.ReactNode => {
```

Comment:
Note: needed to make all the checkboxes in the stories all controlled so the icon displays correctly.

Reply:
Interesting, does it not work with uncontrolled components? I know we had a regression a while back for folks who were using the checkbox component as an uncontrolled component. Did we ensure that functionality is not lost?

Reply:
I don't have context on that... but in storybook when it was uncontrolled the check icon didn't appear - but the background color did change. If this is a regression, it should be fixed in a separate ticket.

Reply:
Probably does need to work in both controlled and uncontrolled situations

Reply:
Ohhhhhh ok I see what was happening now. It seems like it was working in both situations kind of on accident.

Before, the checkbox icon was always visible because the hidden prop doesn't do anything. But it wasn't actually visible because it is the same color as the background when its unchecked. It was supposed to work where the icon is only visible if the checked prop is true.

The BaseRadioCheckboxControl has styles that change the backgroundColor when the sibling element (the hidden checkbox) is active.

Now, that the icon only shows is working as it was supposed to before, when the component is uncontrolled the background color changes as expected but the checked prop is undefined so the icon doesnt appear.

So now, I think I should add a new state or something that tracks the checked state of the hidden checkbox when the checked prop is undefined so the icon can show correctly.